raster-foundry/raster-foundry
# Make pan-to new annotations optional

## Overview

~This PR adds a toggle on the annotation page that allows the user to enable/disable the `panTo` action after an annotation is created. The setting is persisted in local storage.~

This PR makes some changes to improve the annotation UI/UX. It:
- removes the 'pan-to' animation when annotations are created
- moves from hovering to clicking to initiate a 'pan-to' of an existing annotation
- adds dropdown menus both to the annotation page in general and each annotation since the number of options for each was too many to display cleanly
- makes some style changes

### Checklist

### Demo

![image](https://user-images.githubusercontent.com/2442245/45063117-4da5a480-b07a-11e8-8dd4-5ac174aca562.png)


### Notes

~I'm unsure about the styling, the setting feels to prominent for something that will not be toggled very often. Could probably use some design input here @designmatty.~

~This is now WIP because I realized I haven't addressed the hover action...~

Clicking on the button that opens the individual annotation dropdown also pans to the corresponding annotation because of event propagation. I am OK with this and so I did not take any actions to prevent this.

## Testing Instructions

 ~* Make some annotations and see that the pan-to action after creating them respects the state of the toggle~
* Try each option in the annotation page dropdown to ensure each thing works as expected
* Create some annotations, see that each annotations actions work as expected
* Try clicking on annotations to pan to them

Closes #3315
